mininet/link.py

```diff
@@ -90,8 +104,9 @@ def setMAC( self, macstr ):
                  self.ifconfig( 'hw', 'ether', macstr ) +
                  self.ifconfig( 'up' ) )
 
-    _ipMatchRegex = re.compile( r'\d+\.\d+\.\d+\.\d+' )
-    _macMatchRegex = re.compile( r'..:..:..:..:..:..' )
+    _ipMatchRegex = re.compile( r'inet\s+(\d+\.\d+\.\d+\.\d+)\s' )
+    _ip6MatchRegex = re.compile( r'inet6\s+([0-9a-fA-F:\.]+)\s.*scopeid\s.*<(global|link|host)>' )
+    _macMatchRegex = re.compile( r'ether\s+([0-9a-fA-F:]+)\s' )
```

Comment:
The changes here include the following:

- Additional `ifconfig` context given for each item (MAC address or IP address) to more accurately identify it.  For example, if there were an IPv6 address on an interface, the previous MAC address pattern might have picked it up in some cases.
- IPv6 regex added, including scope